### PR TITLE
build: more resources under RBE for certain tests

### PR DESCRIPTION
**Pull request by rickystewart**
Epic: CRDB-8308
Release note: None

### CODE REVIEW

File: pkg/storage/BUILD.bazel [hunk 1]
```diff
@@ -140,6 +140,7 @@ go_test(
     args = ["-test.timeout=295s"],
     data = glob(["testdata/**"]),
     embed = [":storage"],
+    exec_properties = {"Pool": "medium"},
     shard_count = 2,
     tags = ["no-remote-exec"],
```
Comment (healthy-pod) [inline pkg/storage/BUILD.bazel:145]: are you aware of this?

**Review by healthy-pod — APPROVED** [review]

File: pkg/storage/BUILD.bazel (hunk 1 above)
Comment (healthy-pod) [reply pkg/storage/BUILD.bazel:145]: `tags = ["no-remote-exec"],`

Comment (rickystewart) [reply pkg/storage/BUILD.bazel:145]: Yes. Following up from #109176, it seems this test needs more resources. I'll follow up with removing the `no-remote-exec` later.

**Comment by rickystewart** [comment]
bors r=healthy-pod

**Pull request merged**